lib/std/json/static.zig

```diff
@@ -459,7 +473,18 @@ fn internalParse(
                                 return try value_list.toOwnedSliceSentinel(@as(*const u8, @ptrCast(sentinel_ptr)).*);
                             }
                             if (ptrInfo.is_const) {
-                                switch (try source.nextAllocMax(allocator, .alloc_if_needed, options.max_value_len.?)) {
+                                var allocate_when: AllocWhen = .alloc_always;
+                                if (options.allocate) |alloc_when| {
```

Comment:
i suggest implementing the null handling in `parseFromTokenSourceLeaky()` along with the `max_value_len` option. That way, we can avoid the runtime branches in the middle of the processing. then we can just `options.allocate_when.?` in this part of the code.

Reply:
Logic moved, hopefully mirroring `max_value_len` correctly.